src/sql/relation.rs

```diff
@@ -907,6 +907,28 @@ mod tests {
         println!("query = {q}");
     }
 
+    #[test]
+    fn test_reduce_with_only_group_by_columns() {
```

Comment:
Are we sure this is not working now without the change?

Is it working with piled map reduce: R, MR, RM, MRM, RMR